store/sqlstore/channel_store.go

```diff
@@ -595,6 +562,88 @@ func (s SqlChannelStore) migrateMembershipToSidebar(transaction *gorp.Transactio
 	return memberships, nil
 }
 
+func (s SqlChannelStore) migrateFavoritesToSidebarT(transaction *gorp.Transaction, userId, teamId, favoritesCategoryId string) error {
+	if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
```

Comment:
Since we have to do it for MySQL anyway, and the code there is already cross-database compatible, do we need the additional complexity (and testing consideration) here for Postgres?

Reply:
Yeah, that's true. I was thinking that only making 1 database call vs N+1 database calls (where N is the number of favourited channels) would be better, but that's probably an unnecessary optimization.